PSReadLine/Prediction.cs

```diff
@@ -16,8 +18,42 @@ namespace Microsoft.PowerShell
 {
     public partial class PSConsoleReadLine
     {
-        private const string PSReadLine = "PSReadLine";
-        private static PredictionClient s_predictionClient = new(PSReadLine, PredictionClientKind.Terminal);
+        private const string DefaultName = "PSReadLine";
+        private static readonly PredictionClient s_predictionClient = new(DefaultName, PredictionClientKind.Terminal);
+        private static PropertyInfo s_pCurrentLocation = null;
+
+        /// <summary>
+        /// Initialize the <see cref="PropertyInfo"/> objects for those public settable properties newly added to
+        /// <see cref="PredictionClient"/>.
+        /// </summary>
+        private static void InitializePropertyInfo()
+        {
+            Version ver = typeof(PSObject).Assembly.GetName().Version;
+            if (ver.Major < 7 || ver.Minor < 4)
+            {
+                return;
+            }
+
+            Type pcType = typeof(PredictionClient);
+            // Property added in 7.4
+            s_pCurrentLocation = pcType.GetProperty("CurrentLocation");
+        }
+
+        /// <summary>
+        /// New public settable properties may be added to the <see cref="PredictionClient"/> type as it evolves to
+        /// offer more helpful context information. We dynamically set those properties here to avoid any backward
+        /// compatibility issues.
+        /// </summary>
+        private static void UpdatePredictionClient(Runspace runspace, EngineIntrinsics engineIntrinsics)
+        {
+            // Set the current location if the 'CurrentLocation' property exists.
+            if (s_pCurrentLocation is not null)
+            {
+                // Set the current location if it's a local Runspace. Otherwise, set it to null.
+                object path = runspace.RunspaceIsRemote ? null : engineIntrinsics.SessionState.Path.CurrentLocation;
+                s_pCurrentLocation.SetValue(s_predictionClient, path);
```

Comment:
And so because reflection is weird, this is setting the property `CurrentLocation` on the instance `s_predictionClient` to the value of `path`, just like `s_predictionClient.CurrentLocation = path` would, but avoids the issue where that property doesn't exist in previous versions.

Reply:
Yeah, that's correct.